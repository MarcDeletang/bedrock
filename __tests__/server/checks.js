const importChecks = require('../../app/server/checks.ts');
const importBase = require('../base.js');
const { checkParams } = importChecks;
const { throwGenerator, snapGenerator, generateIt } = importBase;

const valid = { init: () => true };

const services = [valid];
const models = [valid];
const registers = [() => true];
const decorations = [{ target: 'request', name: 'test', method: () => true }];
const routes = {};

const checkParamsThrow = throwGenerator(checkParams);

const testsThrows = [
  { m: 'checks invalid services', p: [true] },
  { m: 'checks invalid services', p: [null] },
  { m: 'checks invalid services', p: [{}] },
  { m: 'checks invalid services.init', p: [[{}]] },
  { m: 'checks invalid services.init', p: [[{ init: true }]] },
  { m: 'checks invalid models', p: [[], true] },
  { m: 'checks invalid models', p: [[], null] },
  { m: 'checks invalid models', p: [[], {}] },
  { m: 'checks invalid models.init', p: [[], [{}]] },
  { m: 'checks invalid models.init', p: [[], [{ init: true }]] },
  { m: 'checks invalid registers', p: [[], [], true] },
  { m: 'checks invalid registers', p: [[], [], null] },
  { m: 'checks invalid registers', p: [[], [], {}] },
  { m: 'checks invalid registers.init', p: [[], [], [{}]] },
  { m: 'checks invalid registers.init', p: [[], [], [{ init: true }]] },
];

describe('Testing the server checks', () => {

  it('Check valid empty params', () => {
    expect(checkParams([], [], [], [], {})).toMatchSnapshot();
  });

  testsThrows.map(test => generateIt(test.m, checkParamsThrow, test.p));

  it('Check valid not empty params', () => {
    expect(checkParams(services, models, registers, decorations, routes)).toMatchSnapshot();
  });

});
