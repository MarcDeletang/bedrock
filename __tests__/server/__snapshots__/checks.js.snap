// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the server checks Check valid empty params 1`] = `true`;

exports[`Testing the server checks Check valid not empty params 1`] = `true`;

exports[`Testing the server checks checks invalid models 1`] = `"models"`;

exports[`Testing the server checks checks invalid models 2`] = `"models"`;

exports[`Testing the server checks checks invalid models 3`] = `"models"`;

exports[`Testing the server checks checks invalid models.init 1`] = `"models.init is not a function for [object Object]"`;

exports[`Testing the server checks checks invalid models.init 2`] = `"models.init is not a function for [object Object]"`;

exports[`Testing the server checks checks invalid registers 1`] = `"registers"`;

exports[`Testing the server checks checks invalid registers 2`] = `"registers"`;

exports[`Testing the server checks checks invalid registers 3`] = `"registers"`;

exports[`Testing the server checks checks invalid registers.init 1`] = `"registers is not a function, got: {}"`;

exports[`Testing the server checks checks invalid registers.init 2`] = `"registers is not a function, got: {\\"init\\":true}"`;

exports[`Testing the server checks checks invalid services 1`] = `"services"`;

exports[`Testing the server checks checks invalid services 2`] = `"services"`;

exports[`Testing the server checks checks invalid services 3`] = `"services"`;

exports[`Testing the server checks checks invalid services.init 1`] = `"services.init is not a function for [object Object]"`;

exports[`Testing the server checks checks invalid services.init 2`] = `"services.init is not a function for [object Object]"`;
